Library/Homebrew/diagnostic.rb

```diff
@@ -106,6 +106,34 @@ def please_create_pull_requests(what = "unsupported configuration")
         EOS
       end
 
+      def examine_git_origin(repository_path, desired_origin)
+        return if !Utils.git_available? || !repository_path.git?
+
+        current_origin = repository_path.git_origin
+
+        if current_origin.nil?
+          <<~EOS
+            Missing #{desired_origin} git origin remote.
+
+            Without a correctly configured origin, Homebrew won't update
+            properly. You can solve this by adding the remote:
+              git -C "#{repository_path}" remote add origin #{Formatter.url("https://github.com/#{desired_origin}.git")}
+          EOS
+        elsif current_origin !~ %r{#{desired_origin}(\.git|/)?$}i
+          <<~EOS
+            Suspicious #{desired_origin} git origin remote found.
+
+            With a non-standard origin, Homebrew won't pull updates from
+            the main repository. The current git origin is:
+              #{current_origin}
+
+            Unless you have compelling reasons, consider setting the
```

Comment:
Remove `Unless you have compelling reasons`; people should consider it regardless and their compelling reasons can overrule them 😆